### PR TITLE
Refactor: Plugin `jenkins-pipeline-kubernetes` Refactoring

### PR DESCRIPTION
**Pull request by daniel-hutao**
## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [ ] My code has the necessary comments and documentation (if needed).
- [ ] I have added relevant tests

## Description

Plugin jenkins-pipeline-kubernetes refactoring. I'll add the documents and test this plugin in the next pr.

See commits history for more info.

## Related Issues

#993 

## New Behavior (screenshots if needed)

N/A

### CODE REVIEW

File: internal/pkg/plugin/gitlabcedocker/options.go [hunk 1]
```diff
@@ -57,7 +57,7 @@ func (opts *Options) setGitLabURL() {
 }
 
 func showHelpMsg(options plugininstaller.RawOptions) error {
-	log.Infof("GitLab access URL: %s", gitlabURL)
+	log.Infof("GitLab access JenkinsURL: %s", gitlabURL)
```
Comment (aFlyBird0) [inline internal/pkg/plugin/gitlabcedocker/options.go:60]: I guess you refactored here by mistake.

File: internal/pkg/plugin/jenkinsgithub/options.go [hunk 2]
```diff
@@ -44,7 +44,7 @@ type (
 	// JobOptions is the struct to render job xml.
 	// TODO(aFlyBird0): it seems that GitHubProjectURL and GitSCMURL could be set separately,
 	// now they are set together. Need more investigation.
-	// TODO(aFlyBird0): figure out what does diffent format of GitHub repo URL mean and handle it.
+	// TODO(aFlyBird0): figure out what does diffent format of GitHub repo JenkinsURL mean and handle it.
```
Comment (aFlyBird0) [inline internal/pkg/plugin/jenkinsgithub/options.go:47]: I guess you refactored here by mistake.

File: internal/pkg/plugin/jenkinspipelinekubernetes/update.go [hunk 3]
```diff
@@ -1,26 +1,33 @@
...
+	// Execute all Operations in Operator
+	status, err := operator.Execute(plugininstaller.RawOptions(options))
+	if err != nil {
+		return nil, err
 	}
+	log.Debugf("Return map: %v", status)
+	return status, nil
+}
 
+func Todo(options plugininstaller.RawOptions) error {
```
Comment (aFlyBird0) [inline internal/pkg/plugin/jenkinspipelinekubernetes/update.go:30]: cool

**Review by aFlyBird0 — CHANGES_REQUESTED** [review]

File: internal/pkg/show/config/plugins/jenkins-pipeline-kubernetes.yaml [hunk 4]
```diff
@@ -7,20 +7,13 @@ tools:
...
-      # if all set, devstream will read the password from the config file or outputs from jenkins plugin first, then env var.
-      password:
-      # jenkins job name, mandatory
-      jobName:
-      # path to the pipeline sript file path, relative to the git repo root directory. default: Jenkinsfile
-      pipelineScriptPath: Jenkinsfile
-    # github repo url where the pipeline script is located. mandatory
-    githubRepoUrl: https://github.com/YOUR_GITHUB_ACCOUNT/YOUR_TEST_PROJECT_NAME.git
+    # jenkinsURL
+    jenkinsURL: http://localhost:8080
```
[the remark this answers is not in the document]
Comment (iyear) [reply internal/pkg/show/config/plugins/jenkins-pipeline-kubernetes.yaml:11]: or use `url` tag

https://github.com/go-playground/validator/blob/master/validator_test.go#L7729

File: internal/pkg/plugin/jenkinspipelinekubernetes/update.go (hunk 3 above)
Comment (daniel-hutao) [reply internal/pkg/plugin/jenkinspipelinekubernetes/update.go:30]: thx

**Comment by daniel-hutao** [comment]
@aFlyBird0 @iyear PTAL again, thx!

**Review by iyear — APPROVED** [review]
LGTM 👏

File: internal/pkg/plugin/jenkinspipelinekubernetes/validate.go [hunk 5]
```diff
@@ -24,22 +40,24 @@ func validate(options *Options) []error {
...
-	if options.J.User == "" {
-		options.J.User = defaultJenkinsUser
+	if options.JenkinsUser == "" {
+		options.JenkinsUser = defaultJenkinsUser
 	}
 
-	options.J.URL = "http://" + options.J.URL
+	if !strings.Contains(options.JenkinsURL, "http") {
+		options.JenkinsURL = "http://" + options.JenkinsURL
+	}
```
Comment (iyear) [inline internal/pkg/plugin/jenkinspipelinekubernetes/validate.go:53]: `url` tag do not need this default value set. It must have prefix `http(s)://`

Comment (daniel-hutao) [reply internal/pkg/plugin/jenkinspipelinekubernetes/validate.go:53]: the "defaults" logic is before the "validate".

Comment (daniel-hutao) [reply internal/pkg/plugin/jenkinspipelinekubernetes/validate.go:53]: image

**Review by aFlyBird0 — APPROVED** [review]

**Pull request merged**